sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statsEstimation/UnionEstimation.scala

```diff
@@ -88,9 +88,18 @@ object UnionEstimation {
         case (attrs, outputIndex) =>
           val dataType = unionOutput(outputIndex).dataType
           val statComparator = createStatComparator(dataType)
-          val minMaxValue = attrs.zipWithIndex.foldLeft[(Option[Any], Option[Any])]((None, None)) {
-              case ((minVal, maxVal), (attr, childIndex)) =>
+          val colStatValues = attrs.zipWithIndex
+            .foldLeft[(Option[Any], Option[Any], Option[BigInt])]((None, None, None)) {
+              case ((minVal, maxVal, totalNullCount), (attr, childIndex)) =>
                 val colStat = union.children(childIndex).stats.attributeStats(attr)
+                // Update null count
+                val nullCount = if (totalNullCount.isDefined && colStat.nullCount.isDefined) {
+                  Some(totalNullCount.get + colStat.nullCount.get)
+                } else if (colStat.nullCount.isDefined) {
+                  colStat.nullCount
```

Comment:
If some of `colStat.nullCount`s are missing, we can update the null count in output?

Reply:
Thanks. I updated the code and added a new UT to test the scenario